api/specs/webserver/v0/openapi-share.yaml

```diff
@@ -0,0 +1,62 @@
+openapi: 3.0.0
+info:
+  title: resource sharing API
+  version: 0.1.0
+  description: 'Operations on sharing studies'
+servers:
+  - description: API server
+    url: '/v0'
+paths:
+  /share/study/{study_id}:
+    get:
+      summary: Gets sharing tokens for the given study_id
+      operationId: get_share_study_tokens
+      parameters:
+        - $ref: './openapi-share.yaml#/components/parameters/StudyId'
+      responses:
+        '200':
+          description: got tokens for sharing study
+          content:
+            application/json:
+              schema:
+                $ref: './components/schemas/share_study_tokens.yaml#/ShareStudyTokensEnveloped'
+        default:
+          $ref: './openapi.yaml#/components/responses/DefaultErrorResponse'
+
+  /shared/study/{token_id}:
+    get:
+      summary: Gets shared study for the given token_id
+      operationId: get_shared_study
+      parameters:
+        - $ref: './openapi-share.yaml#/components/parameters/TokenId'
+      responses:
+        '200':
+          description: new study_id
+          content:
+            application/json:
+              schema:
+                $ref: './components/schemas/share_study_tokens.yaml#/SharedStudyIdEnveloped'
+        '302':
+          description: url to be redirected to containing the study ID that was copied/shared
+        default:
+          $ref: './openapi.yaml#/components/responses/DefaultErrorResponse'
```

Comment:
why not using the /projects endpoint with a parameter such as __/project?from_token=__ ? this would be more consistent with the current API.

Reply:
No, here I disagree. You are not getting from your catalog of projects but rather importing into it

``POST  /project/{project_id}/token/{}?action=import``

Reply:
so from the link you posted and others I read there are actually 2 lines of thoughts there:
1. *the google way*:  POST /projects:import with body parameters {token=token_id}
2. *the other way*: POST /projects creates a project , then GET /projects/{id} that returns a response with LINKS to actions such as import, something like ```"Links":[
    {
        "Action":"import",
        "Method":"POST",
        "Href":"http://projects/123",
        "Title":"import a project"
    }}```, then POST /project/123 with body {"Action":"import", "token":token_id}

Personnally prefer version 1...

Reply:
agreed. I misread the details and did an interpretation that followed our style

Reply:
NOTE: there are two types of urls here: ones are sent by email and those need to be a pure url w/o body. Clicking them produces a GET url. The approach there shall be handled independently from the restAPI